### PR TITLE
[DOCS] Update get_context() overload to indicate returned EphemeralDataContext

### PR DESCRIPTION
**Pull request by Rachel-Reverie**
## Description
- Updates get_context() overload to indicate returned EphemeralDataContext

## Definition of done
- [X] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [X] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [X] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [X] Appropriate tests and docs have been updated

### CODE REVIEW

File: great_expectations/data_context/data_context/context_factory.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def get_context(  # noqa: PLR0913
     cloud_organization_id: str | None = ...,
     cloud_mode: bool | None = ...,
     mode: None = ...,
-) -> AbstractDataContext:
+) -> EphemeralDataContext:
```
Comment (Kilo59) [inline great_expectations/data_context/data_context/context_factory.py]:
```suggestion
) -> CloudDataContext | FileDataContext | EphemeralDataContext:
```

I think we actually need a 4th overload to capture just the `EphemeroralDataContext` case, the arguments above do not guarantee an `EphemoralContext` will be returned.

```python
@overload
def get_context(  # noqa: PLR0913
    project_config: DataContextConfig | Mapping | None = ...,
    context_root_dir: None = ...,
    project_root_dir: PathStr | None = ...,
    runtime_environment: dict | None = ...,
    cloud_base_url: None = ...,
    cloud_access_token: None = ...,
    cloud_organization_id: None = ...,
    cloud_mode: None = ...,
    mode: Literal["ephemeral"] = ...,
) -> EphemeralDataContext:
   ...

@overload
def get_context(  # noqa: PLR0913
    project_config: DataContextConfig | Mapping | None = ...,
    context_root_dir: PathStr | None = ...,
    project_root_dir: PathStr | None = ...,
    runtime_environment: dict | None = ...,
    cloud_base_url: str | None = ...,
    cloud_access_token: str | None = ...,
    cloud_organization_id: str | None = ...,
    cloud_mode: bool | None = ...,
    mode: None = ...,
) -> EphemeralDataContext | FileDataContext | CloudDataContext:
   ...

```

**Comment by Kilo59** [comment]
Closing until I have time to fix the typing issues.

**Comment by Kilo59** [comment]
This needs to be cherrypicked to `0.18.x` as well.

File: great_expectations/data_context/data_context/context_factory.py [hunk 2]
```diff
@@ -355,6 +355,21 @@ def _get_ephemeral_context(
...
+    project_config: DataContextConfig | Mapping | None = ...,
+    context_root_dir: None = ...,
+    project_root_dir: None = ...,
+    runtime_environment: dict | None = ...,
+    cloud_base_url: None = ...,
+    cloud_access_token: None = ...,
+    cloud_organization_id: None = ...,
+    cloud_mode: Literal[False] | None = ...,
+    mode: Literal["ephemeral"] = ...,
+) -> EphemeralDataContext:
```
Comment (Kilo59) [inline great_expectations/data_context/data_context/context_factory.py:369]: @cdkini Can you double check these params look right to you?

**Review by cdkini — APPROVED** [review]

**Pull request merged**